### PR TITLE
Pretty sequence indexing

### PR DESCRIPTION
**Pull request by kvark**
This is super useful when inspecting serialized vectors for an ability to quicky jump to an element by index. Looks like this:
```rust
    array: [
        (),// [0]
        (),// [1]
        (),// [2]
    ],
```
Ideally, I'd like to see it like this:
```rust
    array: [
        /*0*/ (),
        /*1*/ (),
        /*2*/ (),
    ],
```
But this is blocked on supporting those types of comments.

### CODE REVIEW

File: src/ser/mod.rs [hunk 1]
```diff
@@ -97,6 +97,7 @@ impl Default for PrettyConfig {
             new_line: "\r\n".to_string(),
             indentor: "    ".to_string(),
             separate_tuple_members: false,
+            enumerate_arrays: true,
```
Comment (ozkriff) [inline src/ser/mod.rs]: Hmmmm, are you sure that this option should be enabled by default?

Comment (kvark) [reply src/ser/mod.rs]: Not really. Would you not want to see the indices?

File: src/ser/mod.rs [hunk 2]
```diff
@@ -418,8 +424,15 @@ impl<'a> ser::SerializeSeq for &'a mut Serializer {
         value.serialize(&mut **self)?;
         self.output += ",";
 
-        if let Some((ref config, _)) = self.pretty {
+        if let Some((ref config, ref mut pretty)) = self.pretty {
+            if config.enumerate_arrays {
+                //TODO: when /**/ comments are supported, prepend the index
+                // to an element instead of appending it.
+                // Note: this doesn't if `new_line` is not an actual new line.
```
Comment (torkleyy) [inline src/ser/mod.rs]: I suggest we assert that `PrettyConfig::new_line` contains a `b'\n'` on creation.

File: src/ser/mod.rs (hunk 1 above)
Comment (torkleyy) [reply src/ser/mod.rs]: I agree with @ozkriff, I think it's too noisy for it to be the default.

**Comment by kvark** [comment]
@torkleyy thanks for feedback! PTAL

**Comment by torkleyy** [comment]
Thank you!

bors r+

**Pull request merged**